cmd/templates-api/error.go

```diff
@@ -113,19 +113,9 @@ func handleError(_ context.Context, w http.ResponseWriter, err error) {
 
 // formatError sets HTTP status code to error.
 func formatError(err error) goaHTTP.Statuser {
-	// Http status.
-	httpCode := 500
-	var httpCodeProvider goaHTTP.Statuser
-	var serviceError *goa.ServiceError
-	if errors.As(err, &httpCodeProvider) {
-		httpCode = httpCodeProvider.StatusCode()
-	} else if errors.As(err, &serviceError) {
-		httpCode = http.StatusBadRequest
-	}
-
 	return errorWithStatusCode{
 		error:    err,
-		httpCode: httpCode,
+		httpCode: errorHttpCode(err),
```

Comment:
V tomto commite sa riesi lepsia integracia s DD. Hodi sa to pre vyvoj/testovanie, vediet, co sa tam deje.

Tento kod sa presunul do funkcie, kedze sa uz pouziva na 2 miestach.